src/Symfony/Component/Messenger/Transport/Receiver/ReceiverInterface.php

```diff
@@ -21,11 +23,42 @@ interface ReceiverInterface
     /**
      * Receive some messages to the given handler.
      *
-     * The handler will have, as argument, the received {@link \Symfony\Component\Messenger\Envelope} containing the message.
+     * The handler will have, as argument, the received {@link \Symfony\Component\Messenger\Envelope}
      * Note that this envelope can be `null` if the timeout to receive something has expired.
+     * containing the message and an instance of @link \Symfony\Component\Messenger\Transport\QueuedMessageMetadata}.
+     * The handler should not throw an exception.
+     *
+     * @throws TransportException If there is an issue communicating with the transport
      */
     public function receive(callable $handler): void;
 
+    /**
+     * Acknowlege that the message was handled (i.e. remove from queue).
+     *
+     * @param mixed $message A message as understood by the transport
+     *
+     * @throws TransportException If there is an issue communicating with the transport
+     */
+    public function acknowledge($message): void;
```

Comment:
Can't we have these methods in another interface instead? Because it doesn't apply to all of them so I'm not sure we should put them here.

Reply:
You're thinking like when a receiver is an API endpoint? Hmm. What might you call that? It's like a `QueueReceiverInterface`??? I'm interested in this...

Reply:
`AsynchronousReceiverInterface`?

Reply:
Actually, it's done! It's not exactly another interface, `ReceiverInterface` now has a `getResponder()` method, which returns an optional object that's able to "respond" back to the queue with ack/nack sort of things.

The reason this couldn't live in a totally independent interface (e.g. so that `AmqpSender implements SenderInterface, QueueReceiverInterface`) is that it killed decoration around `SenderInterface`, like `StopWhenMemoryUsageIsExceededReceiver` because these would only decorate `SenderInterface`... but maybe the sender actually implemented `QueueReceiverInterface` also... hence the, sort of, odd solution.